### PR TITLE
Issue #15214: migrated section 7.1.1 general form to whole config testing

### PR DESCRIPTION
**Pull request by Zopsss**
#15214

### CODE REVIEW

**Review by romani — CHANGES_REQUESTED** [review]
items

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java [hunk 1]
```diff
@@ -1,24 +1,31 @@
-package
-    /** warn */
+package // violation 'package statement should not be line-wrapped.'
+    /** warn. */
```
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java]: I did `valid javadoc.` and `invalid javadoc.` 

https://github.com/checkstyle/checkstyle/pull/15387#discussion_r1693978804 here you said to convert `valid.` to `valid javadoc.` so I thought for `warn` it should be `invalid javadoc.`

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
     com.google.checkstyle.test.chapter7javadoc.rule711generalform;
-// violation 2 lines above 'Javadoc comment is placed in the wrong location.'
 
-/** warn */ // violation 'Javadoc comment is placed in the wrong location.'
+// violation 3 lines above 'Javadoc comment is placed in the wrong location.'
```
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java]: can't, it was  giving `// should be separated from above line`. Moved the violation msg to single line:

https://github.com/checkstyle/checkstyle/blob/46341423888e19e8db4dbde700251d6da13a154a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java#L1-L3

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java [hunk 3]
```diff
@@ -30,18 +37,24 @@ void foo5() {}
...
   /** @inheritDoc */
-  // violation above 'Single-line Javadoc comment should be multi-line.'
+  // 3 violations above:
+  //  'Javadoc tag '@inheritDoc' should be preceded with an empty line.'
+  //  'Single-line Javadoc comment should be multi-line.'
+  //  'Summary javadoc is missing.'
   void foo7() {}
 
-  /** {@inheritDoc} */
+  /** {@inheritDoc}. */
```
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java]: nope, it wasn't required. removed it, my bad

File: src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java [hunk 4]
```diff
@@ -50,117 +63,132 @@ void bar() {}
...
    */
   int bar6() {
     return 0;
   }
 }
 
-/** warn */ // violation 'Javadoc comment is placed in the wrong location.'
-/** valid */
+/** warn. */ // violation 'Javadoc comment is placed in the wrong location.'
+/** valid. */
```
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java]: done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**